sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveExplainSuite.scala

```diff
@@ -207,4 +206,14 @@ class HiveExplainSuite extends QueryTest with SQLTestUtils with TestHiveSingleto
       }
     }
   }
+
+  test("SPARK-28595: explain should not trigger partition listing") {
+    HiveCatalogMetrics.reset()
+    withTable("t") {
+      sql("CREATE TABLE t USING json PARTITIONED BY (j) AS SELECT 1 i, 2 j")
+      assert(HiveCatalogMetrics.METRIC_PARTITIONS_FETCHED.getCount == 0)
+      spark.table("t").explain()
+      assert(HiveCatalogMetrics.METRIC_PARTITIONS_FETCHED.getCount == 0)
```

Comment:
Add a test case that can return non-zero when `spark.sql.legacy.bucketedTableScan.outputOrdering` is set true?